lib/features/order/widgets/order_status_widget.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_multi_app/shared/assets/assets.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class OrderStatusWidget extends StatefulWidget {
+  const OrderStatusWidget({super.key, required this.length, required this.completedSteps});
+  final int length;
+  final int completedSteps;
+
+  @override
+  State<OrderStatusWidget> createState() => _OrderStatusWidgetState();
+}
+
+class _OrderStatusWidgetState extends State<OrderStatusWidget> {
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: List<Widget>.generate(widget.length, (index) {
+        final isCompleted = index < widget.completedSteps;
+        return Padding(
```

Comment:
move to separate widget

Reply:
order status widget already a separated widget